torch/distributed/rpc/api.py

```diff
@@ -63,27 +84,28 @@ def wait_all_workers():
         >>> # do some work
         >>> result = rpc.rpc_sync("worker1", torch.add, args=(torch.ones(1), 1))
         >>> # ready to shutdown
-        >>> rpc.wait_all_workers()
+        >>> rpc.shutdown()
 
         On worker 1:
         >>> import torch.distributed.rpc as rpc
         >>> rpc.init_rpc("worker1", rank=1, world_size=2)
         >>> # wait for worker 0 to finish work, and then shutdown.
-        >>> rpc.wait_all_workers()
+        >>> rpc.shutdown()
     """
     global _agent
-
     if _agent:
-        _agent.join()
-        _agent = None
+        if graceful:
+            _wait_all_workers()
+        _agent.shutdown()
         _destroy_rref_context()
```

Comment:
When you rebase onto #30217, could you please please move this line to the front of `_agent.shutdown()`. This shouldn't make any difference in our tests for now. However, `ctx.destroyed_` is only set to `true` in `_destroy_rref_context()`, meaning that logically, RRef destructions could still trigger messages been sent using the `_agent` before `_destroy_rref_context()`.

Reply:
Should it be before `_wait_all_workers` as well then, since we want to wait for these messages to be processed?

Reply:
Good point! Yes, it should!